test/float16.jl

```diff
@@ -61,6 +61,7 @@ g = Float16(1.)
 @test f*g === Float16(2f0)
 @test f/g === Float16(2f0)
 @test f^g === Float16(2f0)
+@test f^Val{1} === Float16(2f0)
```

Comment:
This test (and the similar tests below) should be removed, since there are (or should be) no `Val` methods for `^`.

To test `literal_pow`, just `@test f^1 === Float16(2f0)`, since `f^1` gets converted into a `literal_pow` call.